android/modules/ui/src/java/ti/modules/titanium/ui/widget/picker/TiUITimePicker.java

```diff
@@ -33,6 +35,9 @@
 	protected Date minDate, maxDate;
```

Comment:
Remove the "suppressChangeEvent" member variable.

Modify the setValue() method to update the "lastSelectedHour" and "lastSelectedMin" member variables before updating the TimePicker object.  This way, when TimePicker.onTimeChanged() method gets invoked, your detectTimeChange() method will not fire the time changed event because the picker's selected time will already match your member variables.